plugins/filters/askAI/index.js

```diff
@@ -104,32 +108,32 @@ const askAI = async (context, role = '', prompt, token, callback) => {
     body: JSON.stringify({
       model: 'gpt-4o-2024-08-06',
       messages: [
-        ...(role ? 
-        [
-          {
-            role: 'system',
-            content: `You are a ${role}. Answer only to the request, without any introductory or conclusion text.`
-          }] 
-          : []),
+        {
+          role: 'system',
+          content: `You are a ${role}. Answer only to the request, without any introductory or conclusion text.`
+        },
         {
           role: 'user',
           content: JSON.stringify(formattedContext)
         },
         { role: 'user', content: prompt }
       ],
-      max_tokens: maxTokens
+      max_tokens: MAX_TOKENS
     })
   });
 
   const data = await response.json();
 
+  if (data?.error?.message) {
+    console.error(data.error.message);
+    return callback(null, data.error.message);
+  }
+
   const suggestion =
     data.choices?.[0]?.message?.content ??
     'context was too big for api, try with smaller context object';
 
-  process.env[cacheKey] = suggestion;
```

Comment:
@MishaKav, don't we need the cache to spare unnecessary calls?

Reply:
we already have it done in higher level. 
every async function will be called only once.